File: docs/snippets/all/descriptors/descr_builtin_component.cpp
#include <rerun.hpp>

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_descriptors_builtin_component");
    rec.spawn().exit_on_failure();

    rec.log_static(
        "data",
        rerun::ComponentBatch::from_loggable(
            rerun::Position3D(1.0f, 2.0f, 3.0f),
            rerun::ComponentDescriptor(
                "user.CustomPoints3D",                            // archetype name
                "user.CustomPoints3D:points",                     // component
                rerun::Loggable<rerun::Position3D>::ComponentType // component type
            )
        )
    );

    // The tags are indirectly checked by the Rust version (have a look over there for more info).
}

File: docs/snippets/all/descriptors/descr_builtin_component.py
#!/usr/bin/env python3

from __future__ import annotations

import rerun as rr  # pip install rerun-sdk

rr.init("rerun_example_descriptors_builtin_component")
rr.spawn()

rr.log(
    "data",
    [
        rr.components.Position3DBatch([1, 2, 3]).described(
            rr.ComponentDescriptor(
                "user.CustomPoints3D:points",
                archetype="user.CustomPoints3D",
                component_type="rerun.components.Position3D",
            )
        )
    ],
    static=True,
)

# The tags are indirectly checked by the Rust version (have a look over there for more info).
